diff --git a/node_modules/zmodem.js/src/zmodem_browser.js b/node_modules/zmodem.js/src/zmodem_browser.js
index 7405f54..32a6c29 100644
--- a/node_modules/zmodem.js/src/zmodem_browser.js
+++ b/node_modules/zmodem.js/src/zmodem_browser.js
@@ -123,7 +123,7 @@ Zmodem.Browser = {
                     };
 
                     reader.onload = function reader_onload(e) {
-                        piece = new Uint8Array(e.target.result, xfer, piece)
+                        piece = new Uint8Array(e.target.result, xfer.get_offset())
 
                         _check_aborted(session);
 
